docs/source/tutorial/bond-extension.rst

```diff
@@ -0,0 +1,130 @@
+.. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Tutorials : How to use the Bond extension package
+#################################################
+
+How to create a bond instrument
+*******************************
+
+There are different types of bonds, which mainly differ in the way the coupon is defined.
+In order to create a bond instrument you first have to decide what type it is.
+The bond extension package currently supports the following bond types:
+
+Fixed rate
+==========
+
+Fixed rate bonds pay a constant coupon each coupon period. The coupon is quoted on a yearly basis (per annum, p.a.), but it could be paid more frequently.
+For example, a bond could have a 2% p.a. coupon and a 6M coupon period. That would mean
+a 1% coupon is paid twice a year.
+
+Here is an example of a bond paying a 1.1% p.a. coupon with a 12M coupon period:
+
+.. code:: daml
+
+  let
+    issueDate = date 2019 Jan 16
+    firstCouponDate = date 2019 May 15
+    maturityDate = date 2020 May 15
+    couponRate = 0.011
+    couponPeriod = M
+    couponPeriodMultiplier = 12
+    redemptionAmount = 1_000_000.0
+  bondInstrument <- originateFixedRateBond custodian issuer "BOND" obs now issueDate holidayCalendarIds calendarDataProvider firstCouponDate maturityDate dayCountConvention businessDayConvention couponRate couponPeriod couponPeriodMultiplier cashInstrumentCid
```

Comment:
I suggest we don't use helper functions like `originateFixedRateBond` (here and below), as those are not available to users.

Reply:
I am now replacing this helper function with the actual code behind it. However, I noticed that `originateFixedRateBond` calls another, more general helper function `createReference` at the end. Are we fine with using that, since it is not bond specific?

Reply:
I wonder if there is a way to hide all usage of `Reference`s from the user, as they are bound to disappear as soon a we have interface keys

Reply:
Alternatively, I could leave the `createReference` part out altogether and refer to the test for more details. I am trying to limit the tutorial to the bond specific topics and to refer to other doc pages (or source code) for the rest, so that it doesn't get too complex.

Reply:
I decided to leave it out for the time being.